situp-plugins/elasticsearch/src/main/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSink.java

```diff
@@ -194,19 +200,49 @@ private long calcEstimatedSizeInBytes(final IndexRequest indexRequest) {
   }
 
   private boolean flushBatch(final BulkRequest bulkRequest) {
+    final BulkResponse bulkResponse;
     try {
-      final BulkResponse bulkResponse = restHighLevelClient.bulk(bulkRequest, RequestOptions.DEFAULT);
-      final boolean hasFailures = bulkResponse.hasFailures();
-      // TODO: apply retry here
-      if (hasFailures) {
-        handleFailures(bulkRequest.requests(), bulkResponse.getItems());
+      bulkResponse = restHighLevelClient.bulk(bulkRequest, RequestOptions.DEFAULT);
```

Comment:
This code has a lot of redundant logic/code. This logic can be made recursive if you treat . The try-catch block looks less readable because we are trying to treat "Success in First attempt" differently from "Success in Second attempt". 


```
ElasticSearchSink.java

bulkRetryStrategy.submit(request);

```


``` 
BulkRetryStrrategy.java

void submit(BulkRequest request) {
    return handleRetry(request, null);
}


void handleRetry(Bulkrequest request, BulkResponse response) {
      Bulkrequest reissueRequest;
      if(response==null) {
           reissueRequest = request;
       }
      else  {
           reissueRequest = buildRetryRequest(request, response);
           if(reissueRequest.isEmpty())
                 return;
      }
      while(retryBackoff.next()){
         try{
               BulkResponse newResponse = sendToES(reissueRequest);
               if(newResponse.hasErrors)
                    handleRetry(reissueRequest, newResponse);
               else
                    return;
         } catch(exception ex) {
               if(canRetry(ex)){
                   handleRetry(reissueRequest, null)
               } else {
                   handleTerminal(reissueRequest);
               }
         }
     }
}

```

Reply:
Let me take some time to go through this.